test/end.ts

```diff
@@ -0,0 +1,1299 @@
+  //   // -- and there is no Vow deficit or surplus
+  //   function testCageCollateralised() public {
+  it('test cage collateralised', async () => {
+    // Ilk memory gold = init_collateral("gold");
+    const gold = await init_collateral('gold');
+    // Usr ali = new Usr(vat, end);
+    // // make a CDP:
+    // address urn1 = address(ali);
+    const urn1 = user1.address;
+    // gold.gemA.join(urn1, 10 ether);
+    await invoke(admin, gold.gemA, 'join', { user: urn1, wad: l2Eth(eth('10')).res });
+    // ali.frob("gold", urn1, urn1, urn1, 10 ether, 15 ether);
+    await frob(user1, 'gold', urn1, urn1, urn1, l2Eth(eth('10')).res, l2Eth(eth('15')).res);
+    // ali's urn has 0 gem, 10 ink, 15 tab, 15 dai
+
+    // global checks:
+    // assertEq(vat.debt(), rad(15 ether));
+    // assertEq(vat.vice(), 0);
+    expect((await vat.call('debt')).debt).to.deep.equal(rad(eth('15').toBigInt()));
+    expect((await vat.call('vice')).vice).to.deep.equal(rad(0n));
+    // // collateral price is 5
+    // gold.pip.poke(bytes32(5 * WAD));
+    await invoke(admin, gold.pip, 'poke', { wut: wad(5n) });
+    // end.cage();
+    await invoke(admin, end, 'cage');
+    // vm.expectEmit(true, true, true, true);
+    // emit Cage("gold");
+    // end.cage("gold");
+    await invoke(admin, end, 'cage_ilk', { ilk: l2String('gold') });
+    // vm.expectEmit(true, true, true, true);
+    // emit Skim("gold", urn1, 3 ether, 15 ether);
+    // end.skim("gold", urn1);
+    await invoke(admin, end, 'skim', { ilk: l2String('gold'), urn: urn1 });
+
+    // // local checks:
+    // assertEq(art("gold", urn1), 0);
+    // assertEq(ink("gold", urn1), 7 ether);
+    // assertEq(vat.sin(address(vow)), rad(15 ether));
+    expect(await art('gold', urn1)).to.deep.equal(l2Eth(0n));
+    expect(await ink('gold', urn1)).to.deep.equal(l2Eth(eth('7')));
+    expect((await vat.call('sin', { u: vow.address })).sin).to.deep.equal(
+      rad(eth('15').toBigInt())
+    );
+
+    // // global checks:
+    // assertEq(vat.debt(), rad(15 ether));
+    // assertEq(vat.vice(), rad(15 ether));
+    expect((await vat.call('debt')).debt).to.deep.equal(rad(eth('15').toBigInt()));
+    expect((await vat.call('vice')).vice).to.deep.equal(rad(eth('15').toBigInt()));
+    // // CDP closing
+    // vm.expectEmit(true, true, true, true);
+    // emit Free("gold", address(ali), 7 ether);
+    // ali.free("gold");
+    // assertEq(ink("gold", urn1), 0);
+    // assertEq(gem("gold", urn1), 7 ether);
+    // ali.exit(gold.gemA, address(this), 7 ether);
+    await invoke(user1, end, 'free', { ilk: l2String('gold') });
+    expect(await ink('gold', urn1)).to.deep.equal(rad(0n));
+    expect(await gem('gold', urn1)).to.deep.equal(uint(eth('7').toBigInt()));
+    await invoke(user1, gold.gemA, 'exit', { user: _admin, wad: l2Eth(eth('7')).res });
+    // vm.warp(block.timestamp + 1 hours);
+    await starknet.devnet.increaseTime(new Date().getTime() * 1000 + 3600);
```

Comment:
Date.getTime returns miliseconds, it should be new Date().getTime()  / 1000

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime